test/integration/newapp_test.go

```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```

Comment:
Our scripts remove temporary files they know about -- those in `$BASETMPDIR`. Placing temporary files _outside_ of the normal place is a mistake and a bug. If they are placed within our configured temp directory, they will be cleaned up.

> Also I want to have ability to run unit tests from my IDE and don't rely on our scripts at all.

You can _also_ do this if you'd like.